server_extensions_api/src/main/java/com/generalbytes/batm/server/extensions/IIdentity.java

```diff
@@ -21,48 +21,249 @@
 import java.util.Date;
 import java.util.List;
 
+/**
+ * Representation of Identity. Identity is your customer, or more specifically, the detail about your customer.
+ * Every transaction is initiated by an Identity, but how you treat that information is determined by your requirements.
+ */
 public interface IIdentity {
 
     int STATE_NOT_REGISTERED = 0;
     int STATE_REGISTERED = 1;
     int STATE_TO_BE_REGISTERED = 2;
     int STATE_PROHIBITED = 3;
     int STATE_ANONYMOUS = 4;
-    int STATE_PROHIBITED_TO_BE_REGISTERED   = 5;
+    int STATE_PROHIBITED_TO_BE_REGISTERED = 5;
+    int STATE_TO_BE_VERIFIED = 6;
 
-    int TYPE_INTERNAL   = 0; //must not be instantiated by extension
-    int TYPE_EXTERNAL   = 1; //this identity was created externally
+    int TYPE_INTERNAL = 0; //must not be instantiated by extension
+    int TYPE_EXTERNAL = 1; //this identity was created externally
 
+    /**
+     * Always true.
+     *
+     * @return {@code true}
+     */
     boolean isNew();
+
+    /**
+     * Generated public ID of identity.
+     *
+     * @return public ID
+     */
     String getPublicId();
+
+    /**
+     * Custom external ID of identity.
+     *
+     * @return external ID
+     */
     String getExternalId();
+
+    /**
+     * Identity state, one of the following:
+     * <ul>
+     *      <li>STATE_NOT_REGISTERED = 0 - "Not Registered"</li>
+     *      <li>STATE_REGISTERED = 1 - "Registered"</li>
+     *      <li>STATE_TO_BE_REGISTERED = 2 - "Awaiting Registration"</li>
+     *      <li>STATE_PROHIBITED = 3 - "Rejected"</li>
+     *      <li>STATE_ANONYMOUS = 4 - "Anonymous"</li>
+     *      <li>STATE_PROHIBITED_TO_BE_REGISTERED = 5 - "Rejected Wants To Be Registered"</li>
+     *      <li>STATE_TO_BE_VERIFIED = 6 - "Awaiting Verification"</li>
```

Comment:
I'm just afraid we will forget to add new constants here, how about `@link`ing them from javadoc?